Allow serverless ingest without specifying a ports annotation

thallgren opened the pull request:
The ability to intercept a workload without a service is built around the `telepresence.getambassador.io/inject-container-ports` annotation, and it was also required to ingest such a workload. This was counterintuitive and the requirement was removed. An ingest doesn't use a port.

thallgren committed:
Ensure that there's a pod-subnet that includes the traffic-manager pod.

The client will use the IP of the traffic-manager pod for DNS, so that
subnet must be included by the traffic-manager, even if the subnets
are set using the `POD_CIDRS` environment variable.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Update module dependencies.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Allow serverless ingest without specifying a ports annotation

The ability to intercept a workload without a service is built around
the `telepresence.getambassador.io/inject-container-ports` annotation,
and it was also required to ingest such a workload. This was
counterintuitive and the requirement was removed. An ingest doesn't use
a port.

Closes #3741

Signed-off-by: Thomas Hallgren <thomas@tada.se>

P0lip commented on go.mod:
```diff
@@ -172,21 +172,22 @@ require (
 
 replace github.com/telepresenceio/telepresence/rpc/v2 => ./rpc
+
+replace github.com/datawire/k8sapi => github.com/thallgren/k8sapi v0.0.0-20241214214806-eba2ff884c63
```
now that k8sapi change is merged, we can update it

thallgren committed:
CI runs using Colima and needs RecursionBlockDuration to be set.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Add note about `dig` command on macOS to the troubleshooting guide.

Signed-off-by: Thomas Hallgren <thomas@tada.se>

thallgren committed:
Make recursion block more fine-grained and add recursionBlockThreads.

Slow systems won't find time to delete, and some use-cases will send
many requests to the same destination in parallel.

Signed-off-by: Thomas Hallgren <thomas@tada.se>
